WPMedia/arc-themes-components
# TMEDIA-675 - Button Component

## Ticket

- [TMEDIA-675](https://arcpublishing.atlassian.net/browse/TMEDIA-675)

## Description

Added a `<Button>` component as per ticket, and added the following additions

* Update the Sass library to have a new mixin to reduce the need to specify all states in the `.scss` file using `component-with-states()`
* Added additional documentation around Accessibility for the component
* Added `classnames` module for handling `className`

## Acceptance Criteria

* A button can be a label only, icon only, label and icon, or label and dual icons
* Variations:
    * Primary (solid) button and reverse - Large, Medium, and Small sizes
    * Secondary (outline) button and reverse - Large, Medium, and Small sizes
    * Default button - defaults to default styling
* A button can render and act as a link if passed as href 
* A button can have a type -- button (default), submit, or reset
* States of Default, Hover, Focus, Disabled, Active


## Test Steps

1. Checkout branch - `git checkout TMEDIA-675-button`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the button storybook documentation
